### PR TITLE
Implement the ydoc js bundle test

### LINKED ISSUE #9929: Implement the Ydoc JS bundle test
The test should ensure that the changes made to JS files are compatible with running the Ydoc on GraalVM.

The test should include running the `ydocServer.js` bundle, and processing a file edit.

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #9929 

Changelog:
- update: customize Ydoc main hostname and port with environment variables
- add: Ydoc initialization test 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

File: lib/java/ydoc-server/src/test/java/org/enso/ydoc/YdocTest.java [hunk 1]
```diff
@@ -45,39 +67,149 @@ public void setup() {
...
+    private static final int SUFFIX_BYTES = 3;
+
+    private static boolean isOk(BufferData data) {
+      return data.readInt16() == 0;
+    }
+
+    private static UUID readUUID(BufferData data) {
+      try {
+        data.skip(data.available() - UUID_BYTES - SUFFIX_BYTES);
+        var uuidString = data.readString(36);
```
Comment (hubertp) [inline lib/java/ydoc-server/src/test/java/org/enso/ydoc/YdocTest.java]: magic constant?

File: build.sbt [hunk 2]
```diff
@@ -1250,14 +1250,15 @@ lazy val `ydoc-server` = project
...
-      "com.github.sbt"       % "junit-interface"             % junitIfVersion            % Test
+      "org.graalvm.truffle"        % "truffle-api"                 % graalMavenPackagesVersion % "provided",
+      "org.graalvm.polyglot"       % "inspect"                     % graalMavenPackagesVersion % "runtime",
+      "org.graalvm.polyglot"       % "js"                          % graalMavenPackagesVersion % "runtime",
+      "org.slf4j"                  % "slf4j-api"                   % slf4jVersion,
+      "io.helidon.webclient"       % "helidon-webclient-websocket" % helidonVersion,
+      "io.helidon.webserver"       % "helidon-webserver-websocket" % helidonVersion,
+      "junit"                      % "junit"                       % junitVersion              % Test,
+      "com.github.sbt"             % "junit-interface"             % junitIfVersion            % Test,
+      "com.fasterxml.jackson.core" % "jackson-databind"            % jacksonVersion            % Test
```
Comment (JaroslavTulach) [inline build.sbt:1261]: Adding test only dependency is OK.

File: lib/java/ydoc-server/src/main/java/org/enso/ydoc/Main.java [hunk 3]
```diff
@@ -15,7 +18,19 @@ public static void main(String[] args) throws Exception {
 
     Sampling.init();
 
-    try (var ydoc = Ydoc.builder().build()) {
+    var ydocHost = System.getenv(YDOC_HOST);
+    var ydocPort = System.getenv(YDOC_PORT);
```
Comment (JaroslavTulach) [inline lib/java/ydoc-server/src/main/java/org/enso/ydoc/Main.java]: Why do we define these environment variables, when they are not used anywhere?

File: lib/java/ydoc-server/src/test/java/org/enso/ydoc/YdocTest.java [hunk 4]
```diff
@@ -45,39 +67,149 @@ public void setup() {
...
+
+    @Override
+    public void onMessage(WsSession session, String text, boolean last) {
+      log.error("Got unexpected message [{}].", text);
+    }
   }
 
-  private static final class WebServerWsListener implements WsListener {
-    WebServerWsListener() {}
+  private static final class LanguageServerConnection implements WsListener {
```
Comment (JaroslavTulach) [inline lib/java/ydoc-server/src/test/java/org/enso/ydoc/YdocTest.java:143]: Is this a mock server, mocking language server protocol. The real Ydoc code is connecting to it and we are verifying it connects and exchanges few messages, rigth?

**Review by JaroslavTulach — APPROVED** [review]

File: lib/java/ydoc-server/src/main/java/org/enso/ydoc/Main.java (hunk 3 above)
Comment (4e6) [reply lib/java/ydoc-server/src/main/java/org/enso/ydoc/Main.java]: `Main` class exists only for development purposes. I.e. you can start the Ydoc server on a custom socket with
```sh
$ env YDOC_HOST=127.0.0.1 YDOC_PORT=30030 sbt ydoc-server/run
```

File: lib/java/ydoc-server/src/test/java/org/enso/ydoc/YdocTest.java (hunk 4 above)
Comment (4e6) [reply lib/java/ydoc-server/src/test/java/org/enso/ydoc/YdocTest.java:143]: Correct

File: lib/java/ydoc-server/src/main/java/org/enso/ydoc/Main.java (hunk 3 above)
Comment (hubertp) [reply lib/java/ydoc-server/src/main/java/org/enso/ydoc/Main.java]: Preferably prefixed with `ENSO_`?

**Review by hubertp — APPROVED** [review]

**Pull request merged**